pkg/extensions/brokers/natsjetstream/natsjetstream.go

```diff
@@ -101,38 +128,51 @@ func WithNakDelay(duration time.Duration) ControllerOption {
 	}
 }
 
-// NewController creates a new NATS JetStream controller.
-func NewController(url string, options ...ControllerOption) (*Controller, error) {
-	// Connect to NATS
-	nc, err := nats.Connect(url)
-	if err != nil {
-		return nil, fmt.Errorf("could not connect to nats: %w", err)
+// WithConnectionOpts set the nats.Options to connect to nats.
+func WithConnectionOpts(opts ...nats.Option) ControllerOption {
+	return func(controller *Controller) error {
+		nc, err := nats.Connect(controller.url, opts...)
+		if err != nil {
+			return fmt.Errorf("could not connect to nats: %w", err)
+		}
+		controller.natsConn = nc
+		return nil
 	}
+}
 
-	// Create a JetStream management interface
-	js, err := jetstream.New(nc)
+// registerConsumer set consumer configuration for creates or updates a consumer based on the given configuration
+// at the broker.
+func (c *Controller) registerConsumer() error {
+	if c.consumerConfig.Name == "" {
+		return fmt.Errorf("consumer name is required")
+	}
+	c.consumerName = c.consumerConfig.Name
+
+	_, err := c.jetStream.CreateOrUpdateConsumer(context.Background(), c.streamName, c.consumerConfig)
 	if err != nil {
-		return nil, fmt.Errorf("could not connect to jetstream: %w", err)
+		return fmt.Errorf("could not create or update consumer: %w", err)
 	}
 
-	// Creates default controller
-	controller := &Controller{
-		natsConn:       nc,
-		jetStream:      js,
-		logger:         extensions.DummyLogger{},
-		channels:       make(map[string]chan jetstream.Msg),
-		consumeContext: nil,
-		nakDelay:       time.Second * 5,
+	return nil
+}
+
+// registerStream creates or updates a stream based on the given configuration at the broker.
+func (c *Controller) registerStream() error {
```

Comment:
There is a problem here, examples are broken because it should be possible to use Stream without creating them.
But this change made it mandatory, which will break some usages.

We should put back the stream registration as an option :)

Reply:
Ah yes, sorry. I had to move it from the `WithStreamConfig` and `WithConsumerConfig` function, because register a stream or a consumer requires a fully configured connection to execute. If the broker is secured with TLS and or Auth, we would necessarily require `WithConnectionOpts` to be executed beforehand. I wanted to avoid that.

Reply:
No problem ☺️